kratos/processes/generic_find_elements_neighbours_process.h

```diff
@@ -0,0 +1,228 @@
+#include "utilities/parallel_utilities.h"
+
+
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+typedef  ModelPart::NodesContainerType NodesContainerType;
+typedef  ModelPart::ElementsContainerType ElementsContainerType;
+typedef Geometry<Node < 3 > > GeometryType;
+
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+class  KRATOS_API(KRATOS_CORE) GenericFindElementalNeighboursProcess
+    : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of GenericFindElementalNeighboursProcess
+    KRATOS_CLASS_POINTER_DEFINITION(GenericFindElementalNeighboursProcess);
+
+    typedef GlobalPointersVector<Element> ElementPointerVector;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    GenericFindElementalNeighboursProcess(ModelPart& model_part)
+        : mr_model_part(model_part)
+    {
+    }
+
+    /// Destructor.
+    ~GenericFindElementalNeighboursProcess() override
+    {
+    }
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void ExecuteInitialize() override;
+    
+
+    std::vector<bool> HasNeighboursInFaces(const Element&) ;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "FindElementalNeighboursProcess";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "FindElementalNeighboursProcess";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ModelPart& mr_model_part;
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+
+    GlobalPointer<Element> CheckForNeighbourElems (const Geometry<Node<3> >& rBoundaryGeom,
+                                                   Element & rElement);
+
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+
+    
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+
+
+    ///@}
+    ///@name Private Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    /// Assignment operator.
+    GenericFindElementalNeighboursProcess& operator=(GenericFindElementalNeighboursProcess const& rOther);
```

Comment:
This must be in a `private` section. Note that it is now protected as the `private` keyword is missing.